### PR TITLE
Bump Smallrye RM version from 4.9.0 to 4.10.1

### PR DESCRIPTION
**Pull request by ozangunalp**
- Doc update for Kafka guide
- Auto serde detection for `MultiSplitter` type
- ITs updating using multiple outgoings

### CODE REVIEW

File: bom/application/pom.xml [hunk 1]
```diff
@@ -341,6 +341,15 @@
                 <type>pom</type>
             </dependency>
 
+            <!-- Smallrye Reactive Messaging dependencies, imported as a BOM -->
+            <dependency>
+                <groupId>io.smallrye.reactive</groupId>
```
Comment (cescoffier) [inline bom/application/pom.xml:346]: Not a great idea. Complexify the productization.

**Review by cescoffier — CHANGES_REQUESTED** [review]
Avoid the BOM.
Also, CI failures are related.

File: bom/application/pom.xml (hunk 1 above)
Comment (ozangunalp) [reply bom/application/pom.xml]: Ok I rolled back, we can hold the discussion later

**Comment by ozangunalp** [comment]
Fixed the related `DefaultSerdeConfigTest` failures

**Review by cescoffier — APPROVED** [review]

**Pull request merged**